### ISSUE TITLE
s3 tag key "service_name" being forcefully rewritten

### ISSUE THREAD

**Issue by repudi8or** [pulumi/pulumi-aws#1385]
I am creating an s3 bucket with the tag key "service_name" equal to tag value "my_service" and the end result is tagkey/value "serviceName"="my_service"

This seems to relate to the code in `https://github.com/pulumi/pulumi-aws/blob/fd4aa86ca3b26b301a70fd4cccb7931afc4708bc/sdk/python/pulumi_aws/_tables.py#L5` (SNAKE_TO_CAMEL_CASE_TABLE).... 

Please stop rewriting s3 tag keys. This is breaking parts of my infrastructure.

Reproducible code snippet
`import pulumi
import pulumi_aws as aws

properties_dict = { "properties": {'acl': 'private', 'lifecycle_rules': [{'id': 'incomplete-mpu', 'enabled': True, 'abort_incomplete_multipart_upload_days': 7}], 'server_side_encryption_configuration': None, 'versioning': {'enabled': True}, 'tags': {'service_name': 'my_bucket', 'created_by': 'pulumi'}, 'bucket': 'my_bucket', 'policy': '{"Version": "2012-10-17", "Statement": [{"Sid": "MakeItPublic", "Effect": "Allow", "Principal": "*", "Action": "s3:GetObject", "Resource": "*"}]}'}}
my_provider = aws.Provider('us-east-1', region='us-east-1')
bucket = aws.s3.Bucket("mybucket", **properties_dict['properties'], opts=pulumi.ResourceOptions(provider=my_provider))`

this will result in 'mybucket' created with tag serviceName=my_bucket (instead of service_name=my_bucket as expected)

**Comment by justinvp** [comment]
This is an unfortunate side effect of https://github.com/pulumi/pulumi/issues/3151. We're planning to address the behavior in the next major version of Pulumi, but in the meantime, it's possible to workaround this by creating a subclass of `Bucket` and providing an implementation of `translate_input_property` like discussed in this comment: https://github.com/pulumi/pulumi/issues/3151#issuecomment-704509706

```python
class MyBucket(aws.s3.Bucket):
    def translate_output_property(self, prop):
        if prop in ['service_name']: return prop
        return super().translate_output_property(prop)

    def translate_input_property(self, prop):
        if prop in ['service_name']: return prop
        return super().translate_input_property(prop)
```

**Comment by repudi8or** [comment]
thanks that workaround was ok for s3, but i see the same issue with a bunch of other resource types like kms keys and acm certs... is it possible to fix this at a higher level for all resource types ?

**Comment by leezen** [comment]
Yes, this is tracked in https://github.com/pulumi/pulumi/issues/6199

**Comment by justinvp** [comment]
This issue has been addressed and is available for preview in the pulumi-aws 4.0.0-beta.2 release.
